### PR TITLE
[feat] Rust circuit

### PR DESCRIPTION
**Pull request by ytham**
- Rust circuit (WIP)
- Needs test still

### CODE REVIEW

File: app/kwenta-circuit/src/claim.rs [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    static ref CONDITIONAL_ORDER_EXECUTED_SCHEMA: Option<H256> = H256::from_str("0x3f4c4edf80aee6ea6f4fb3aed498a467e30ed1482bc06539ffe892cf7304e334").ok();
+    static ref KWENTA_SMART_MARGIN_V3_ADDR: Option<Address> = Address::from_str("0xe331a7eeC851Ba702aA8BF43070a178451d6D28E").ok();
+}
+
+#[AxiomComputeInput]
+pub struct ClaimInput {
+    pub block_numbers: FixLenVec<usize, MAX_INPUTS>,
+    pub tx_idxs: FixLenVec<usize, MAX_INPUTS>,
+    pub log_idxs: FixLenVec<usize, MAX_INPUTS>,
+    pub executor_fees: FixLenVec<usize, MAX_INPUTS>,
```
Comment (yi-sun) [inline app/kwenta-circuit/src/claim.rs]: Is this extraneous?

File: app/kwenta-circuit/src/claim.rs [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        for i in 0..MAX_INPUTS {
+            // Validate event contract is equivalent to Kwenta contract
+            let event_contract = api
+                .get_receipt(assigned_inputs.block_numbers[i], assigned_inputs.tx_idxs[i])
+                .log(assigned_inputs.log_idxs[i])
+                .address();
+            let event_contract = api.from_hi_lo(event_contract);
+            let event_contract_is_equal = gate.is_equal(api.ctx(), event_contract, kwenta_contract);
+            let is_out_of_range = gate.is_zero(api.ctx(), in_range[i]);
+            let event_contract_or_zero = gate.or(api.ctx(), event_contract_is_equal, is_out_of_range);
```
Comment (yi-sun) [inline app/kwenta-circuit/src/claim.rs]: Should this be `event_contract_or_one`?

File: app/kwenta-circuit/src/claim.rs [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+                .get_receipt(assigned_inputs.block_numbers[i], assigned_inputs.tx_idxs[i])
+                .log(assigned_inputs.log_idxs[i])
+                .data(two, *CONDITIONAL_ORDER_EXECUTED_SCHEMA);
+            let account_id = account_id_market_id_hilo.hi();
+            if i == 0 {
+                logged_account_id = account_id;
+            } else {
+                let account_id_is_equal = gate.is_equal(api.ctx(), account_id, logged_account_id);
+                let is_out_of_range = gate.is_zero(api.ctx(), in_range[i]);
+                let account_id_or_zero = gate.or(api.ctx(), account_id_is_equal, is_out_of_range);
```
Comment (yi-sun) [inline app/kwenta-circuit/src/claim.rs]: `account_id_or_one`

File: app/kwenta-circuit/src/claim.rs [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+        let last_idx = gate.sub(api.ctx(), assigned_inputs.num_claims, one);
+        let last_claim_id = gate.select_from_idx(
+            api.ctx(),
+            claim_ids,
+            last_idx,
+        );
+        vec![
+            first_claim_id.into(),
+            last_claim_id.into(),
+
```
Comment (yi-sun) [inline app/kwenta-circuit/src/claim.rs]: `account_id` missing?

**Review by yi-sun — CHANGES_REQUESTED** [review]

**Comment by ytham** [comment]
@yi-sun ready for re-review

**Review by yi-sun — APPROVED** [review]
LGTM!

**Pull request merged**